website/sales/api/v2/serializers/shift.py

```diff
@@ -0,0 +1,53 @@
+from rest_framework import serializers
+
+from sales.models.product import ProductListItem
+from sales.models.shift import Shift
+
+
+class ProductListItemSerializer(serializers.ModelSerializer):
+    """Serializer for product list items"""
+
+    class Meta:
+        model = ProductListItem
+        fields = ("name", "price", "age_restricted")
+        read_only_fields = ("name", "price", "age_restricted")
+
+    name = serializers.SerializerMethodField("_name")
+    age_restricted = serializers.SerializerMethodField("_age_restricted")
+
+    def _name(self, instance):
+        return instance.product.name
+
+    def _age_restricted(self, instance):
+        return instance.product.age_restricted
+
+
+class ShiftSerializer(serializers.ModelSerializer):
+    """Serializer for shifts."""
+
+    class Meta:
+        model = Shift
+        fields = (
+            "pk",
+            "start_date",
+            "end_date",
+            "products",
+            "total_revenue",
+            "num_orders",
+            "orders",
+        )
+
+    products = ProductListItemSerializer(
+        source="product_list.product_items", many=True, read_only=True
+    )
+
+    total_revenue = serializers.DecimalField(
+        max_digits=6, decimal_places=2, read_only=True
+    )
+
+    num_orders = serializers.SerializerMethodField("_num_orders")
+
+    def _num_orders(self, instance):
+        return instance.orders.count()
+
+    orders = serializers.HyperlinkedIdentityField(view_name="shift-orders")
```

Comment:
I am not at all planning to add hyperlinks in any v2 API. Because:
1. Everything is identified by a primary key
2. The documentation is available to everyone
Therefore I am not sure I would like to see this in this part of the API.